nautobot/extras/models/metadata.py

```diff
@@ -140,6 +143,23 @@ def delete(self, *args, **kwargs):
         super().delete(*args, **kwargs)
 
 
+class ObjectMetadataManager(BaseManager.from_queryset(RestrictedQuerySet)):
+    use_in_migrations = True
+
+    def get_for_model(self, model):
+        """Return all ObjectMetadatas assigned to the given model."""
+        concrete_model = model._meta.concrete_model
+        cache_key = f"{self.get_for_model.cache_key_prefix}.{concrete_model._meta.label_lower}"
+        queryset = cache.get(cache_key)
+        if queryset is None:
+            content_type = ContentType.objects.get_for_model(concrete_model)
+            queryset = self.get_queryset().filter(content_types=content_type)
+            cache.set(cache_key, queryset)
+        return queryset
```

Comment:
This gives us all of the metadata for a given model *class* (e.g. `Device`) - I'm not sure how useful that will be, I was thinking it'd be for a given *instance* (e.g. `device1`).

Reply:
Implemented